sailor/sap_iot/write.py

```diff
@@ -68,22 +70,82 @@ def _upload_data_single_indicator_group(dataset, indicator_set, group_id, templa
         _upload_data_single_equipment(data_subset, equipment.id, tags)
 
 
-def upload_indicator_data(dataset: TimeseriesDataset):
+def _check_indicator_group_is_complete(uploaded_indicators, indicator_group_id, template_id):
+    missing_indicators = []
```

Comment:
without the loop you don't need this anymore I think. Just assign the list comprehension in line 98 directly instead of the `extend`.

Reply:
when removed error occurs name 'missing_indicators' is not defined

Reply:
that seems very surprising. did you replace the `extend` below by a straight assignment?
i.e. remove this line, and the below, replace the `extend`-line by
`missing_indicators = [x['internalId'] for x in indicator_group['indicators'] if x['id'] not in uploaded_indicators]`
I don't see any use of `missing_indicators` between those two lines?

Reply:
that works, thanks